test/e2e/scrapeconfig_test.go

```diff
@@ -4103,3 +4106,148 @@ var DockerSwarmSDTestCases = []scrapeCRDTestCase{
 		expectedError: false,
 	},
 }
+
+var HetznerSDTestCases = []scrapeCRDTestCase{
+	{
+		name: "Valid Minimal Config",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			HetznerSDConfigs: []monitoringv1alpha1.HetznerSDConfig{
+				{
+					Role: "Hcloud",
+				},
+			},
+		},
+		expectedError: false,
+	},
+	{
+		name: "Invalid Role",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			HetznerSDConfigs: []monitoringv1alpha1.HetznerSDConfig{
+				{
+					Role: "Server",
```

Comment:
```suggestion
					Role: "",
```
Hi @nutmos maybe change the invalid to empty string